test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_core_test.clj

```diff
@@ -174,3 +177,126 @@
+        ;; and I didn't see any other way to pull the queue back from
+        ;; the thread pool.
+        get-queue-method (-> thread-pool
+                             (.getClass)
+                             (.getDeclaredMethod "getQueue" nil))
+        _                (.setAccessible get-queue-method true)
+        queue            (.invoke get-queue-method thread-pool nil)]
+    {:thread-pool thread-pool
+     :queue queue}))
+
+(deftest create-server-tests
+  (testing "thread pool given proper values"
+    (testing "max threads passed through"
+      (dotimes [x 5]
+        (let [{:keys [thread-pool]}
+          (get-thread-pool-and-queue-for-create-server x 1)]
+          (is (= x (.getMaxThreads thread-pool))
+              "Unexpected max threads for queue"))))
+
+    (testing "default idle timeout passed through"
+      (let [{:keys [thread-pool]}
+             (get-thread-pool-and-queue-for-create-server 42 1)]
+        (is (= jetty/default-queue-idle-timeout
+               (.getIdleTimeout thread-pool)))))
+
+    (testing "when queue-max-size less than default-queue-min-threads"
+      (let [max-threads    23
+            queue-min-size (dec jetty/default-queue-min-threads)
+            {:keys [thread-pool queue]}
+              (get-thread-pool-and-queue-for-create-server max-threads
+                                                           queue-min-size)]
+        (is (= max-threads (.getMaxThreads thread-pool))
+            "Unexpected max threads for thread pool")
+        (is (= queue-min-size (.getMinThreads thread-pool))
+            "Unexpected min threads for queue")
+        (is (= queue-min-size (.getCapacity queue))
+            "Unexpected initial capacity for queue")
+        (is (= queue-min-size (.getMaxCapacity queue))
+            "Unexpected max capacity for queue")))
+
+    (testing "when queue-max-size greater than default-queue-min-threads"
+      (let [max-threads    42
+            queue-min-size (inc jetty/default-queue-min-threads)
+            {:keys [thread-pool queue]}
+            (get-thread-pool-and-queue-for-create-server max-threads
+                                                         queue-min-size)]
+        (is (= max-threads (.getMaxThreads thread-pool))
+            "Unexpected max threads for thread pool")
+        (is (= jetty/default-queue-min-threads (.getMinThreads thread-pool))
+            "Unexpected min threads for queue")
+        (is (= jetty/default-queue-min-threads (.getCapacity queue))
+            "Unexpected initial capacity for queue")
+        (is (= queue-min-size (.getMaxCapacity queue))
+            "Unexpected max capacity for queue")))))
+
+(defn get-responses
+  [num-requests port]
+  (doall (for [_ (range num-requests)]
+           (future
+             (http-sync/get
+               (str "http://localhost:"
+                    port
+                    "/hello")
+               {:as :text})))))
+
+(deftest queue-max-size-tests
```

Comment:
The fragility of these tests proved out in Travis runs.

On one Travis cell, https://travis-ci.org/puppetlabs/trapperkeeper-webserver-jetty9/jobs/46255778,  the tests failed because the excessive number of http clients being created by the test caused Java OutOfMemoryErrors.  These may have been possible to overcome by bumping up the Java heap when running tests or manipulating `max-threads` on the Jetty server to be lower such that the number of concurrent requests which needed to be run were much lower.

On another Travis cell, https://travis-ci.org/puppetlabs/trapperkeeper-webserver-jetty9/jobs/46255777, the "of sufficiently low queue-max-size" test failed due to all of the requests actually succeeding despite the lower queue size set.

The particularly challenging aspect of these tests is to generate enough request load on the server that the queue is completely filled while still making these tests more "end-to-end" -- avoiding having to manipulate internal Jetty state and implementation details -- like how many requests are backed up at the different layers - selectors, acceptors, queue, worker threads, ...

I meekly removed this test block in b4da828.  Unit-level tests around setting of the queue size are still in the PR.